### PR TITLE
adds an error message to movables not being removed from the grid... again

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
I swear i didnt fail at this like 3 times i tested it this time.

adds a descriptive error of what spatial grid cells a movable is stuck in, and in what channels. This only runs during unit tests. hopefully this should be enough information to go off of to fix the spurious cockroach error. if its not then i can try tracking all grid cell changes during unit tests. 
error looks like this:
```
[2023-05-03 04:16:34.009] runtime error: /mob/living/trolls_the_maintainer instance, which is in nullspace, and thus not be within the contents of any spatial grid cell, was in the contents of 2 spatial grid cells when it was only supposed to be in one! within the contents of the following cells: {(221, 119, 11), within channels: hearing}, {coords: (136, 136, 14), within channels: hearing}. (code/controllers/subsystem/spatial_gridmap.dm:581)
```
for something located in nullspace but still in the contents of >0 cells and:
```
runtime error: /mob/living/trolls_the_maintainer instance, which is supposed to only be in the contents of a spatial grid cell at coords: (136, 136, 14), was in the contents of 6 spatial grid cells when it was only supposed to be in one! within the contents of the following cells: {(68, 153, 2), within channels: hearing}, {coords: (221, 170, 3), within channels: hearing}, {coords: (255, 153, 11), within channels: hearing}, {coords: (170, 238, 13), within channels: hearing}, {coords: (204, 119, 14), within channels: hearing}, {coords: (136, 136, 14), within channels: hearing}. 
```
if its not in nullspace but its within more than 1 grid cell. 

the coordinates here are translated from the index of the given cell to world coordinates. 
## Why It's Good For The Game
mothblocks has been standing outside my house for weeks i am fearing for my life

### CODE REVIEW

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 1]
```diff
@@ -482,21 +482,113 @@ SUBSYSTEM_DEF(spatial_grid)
...
+		/**
+		 * example:
+		 *
+		 * /mob/living/trolls_the_maintainer instance, which is supposed to only be in the contents of a spatial grid cell at coords: (136, 136, 14),
+		 * was in the contents of 3 spatial grid cells when it was only supposed to be in one! within the contents of the following cells:
+		 * {(68, 153, 2), within channels: hearing},
+		 * {coords: (221, 170, 3), within channels: hearing},
+		 * {coords: (255, 153, 11), within channels: hearing},
+		 * {coords: (136, 136, 14), within channels: hearing}.
+		 * ping @KylerAce in discord if you see this
```
Comment (JohnFulpWillard) [inline code/controllers/subsystem/spatial_gridmap.dm:549]: PLAYER
REFERENCE
:red_square: :blue_square: :red_square: :blue_square: :red_square: :blue_square:

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm]: eh yeah not worth it tbh

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 2]
```diff
@@ -482,21 +482,108 @@ SUBSYSTEM_DEF(spatial_grid)
 
 	return TRUE
 
-///find the cell this movable is associated with and removes it from all lists
-/datum/controller/subsystem/spatial_grid/proc/force_remove_from_cell(atom/movable/to_remove, datum/spatial_grid_cell/input_cell)
+/// if for whatever reason this movable is "untracked" e.g. it breaks the assumption that a movable is only inside the contents of any grid cell associated with its loc,
+/// this will error. this checks every grid cell in the world so dont call this on live unless you have to.
+/// returns TRUE if this movable is untracked, FALSE otherwise
+/datum/controller/subsystem/spatial_grid/proc/untracked_movable_error(atom/movable/movable_to_check)
+
```
Comment (Mothblocks) [inline code/controllers/subsystem/spatial_gridmap.dm:489]:
```suggestion
```

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 3]
```diff
@@ -482,21 +482,108 @@ SUBSYSTEM_DEF(spatial_grid)
...
+	if(!initialized)
+		return FALSE
+
+	var/datum/spatial_grid_cell/loc_cell = get_cell_of(movable_to_check)
+	var/list/containing_cells = find_hanging_cell_refs_for_movable(movable_to_check, FALSE)
+	//if we're in multiple cells, throw an error.
+	//if we're in 1 cell but it cant be deduced by our location, throw an error.
+	if(length(containing_cells) > 1 || (length(containing_cells) == 1 && loc_cell && containing_cells[1] != loc_cell && containing_cells[1] != null))
+
```
Comment (Mothblocks) [inline code/controllers/subsystem/spatial_gridmap.dm:501]:
```suggestion
```

File: code/controllers/subsystem/spatial_gridmap.dm [hunk 4]
```diff
@@ -482,21 +482,108 @@ SUBSYSTEM_DEF(spatial_grid)
...
+/datum/controller/subsystem/spatial_grid/proc/untracked_movable_error(atom/movable/movable_to_check)
+
+	if(!movable_to_check?.spatial_grid_key)
+		return FALSE
+
+	if(!initialized)
+		return FALSE
+
+	var/datum/spatial_grid_cell/loc_cell = get_cell_of(movable_to_check)
+	var/list/containing_cells = find_hanging_cell_refs_for_movable(movable_to_check, FALSE)
```
Comment (Mothblocks) [inline code/controllers/subsystem/spatial_gridmap.dm:497]: Everywhere you call `find_hanging_cell_refs_for_movable`, use a named parameter for TRUE/FALSE

Comment (Kylerace) [reply code/controllers/subsystem/spatial_gridmap.dm]: done'd

**Pull request merged**